gamedata/alldefs_post.lua

```diff
@@ -238,8 +238,41 @@ function UnitDef_Post(name, uDef)
 			end
 		end
 
+		--normal commander respawning
+		if modOptions.comrespawn == "all" then
+			if name == "armcom" or name == "corcom" or name == "legcom" then
+				uDef.customparams.effigy = "comeffigylvl1"
+				uDef.customparams.respawn_condition = "health"
+				uDef.customparams.respawn_announcement = "A Commander Effigy was sacrificed."
+				uDef.customparams.respawn_announcement_size = 18.5
+				uDef.customparams.minimum_respawn_stun = 5
+				uDef.customparams.distance_stun_multiplier = 1
+				local numBuildoptions = #uDef.buildoptions
+				uDef.buildoptions[numBuildoptions + 1] = "comeffigylvl1"
+			end
+		end
+		
+
 		if modOptions.evocom then	
 			if uDef.customparams.isevocom or name == "armcom" or name == "corcom" or name == "legcom" then
+				
+				if modOptions.comrespawn == "all" or modOptions.comrespawn == "evocom" then--add effigy respawning, if enabled
+					uDef.customparams.respawn_condition = "health"
+					
+					Spring.Echo("flarf 1")
```

Comment:
Remove debug echoes

Reply:
Yeah, I'm debugging stuff still. I'll make sure to do a final pass. Thanks.

Reply:
it's gone